src/Tickets/Commerce/Order_Modifiers/Custom_Tables/Abstract_Custom_Table.php

```diff
@@ -0,0 +1,132 @@
+<?php
+/**
+ * Abstract Custom Table class for handling common operations on custom tables, such as adding indexes
+ * and foreign key constraints.
+ *
+ * @since TBD
+ *
+ * @package TEC\Tickets\Commerce\Order_Modifiers\Custom_Tables
+ */
+
+namespace TEC\Tickets\Commerce\Order_Modifiers\Custom_Tables;
+
+use TEC\Common\StellarWP\Schema\Tables\Contracts\Table;
+use wpdb;
+
+/**
+ * Abstract class that provides utility methods for managing custom table schemas,
+ * including adding indexes and foreign key constraints.
+ *
+ * @since TBD
+ *
+ * @package TEC\Tickets\Commerce\Order_Modifiers\Custom_Tables
+ */
+abstract class Abstract_Custom_Table extends Table {
+
+	/**
+	 * Helper method to check and add an index to a table.
+	 *
+	 * @since TBD
+	 *
+	 * @param wpdb   $wpdb       The WordPress database global.
+	 * @param array  $results    The results array to track changes.
+	 * @param string $table_name The name of the table.
+	 * @param string $index_name The name of the index.
+	 * @param string $columns    The columns to index.
+	 *
+	 * @return array The updated results array.
+	 */
+	protected function check_and_add_index( wpdb $wpdb, array $results, string $table_name, string $index_name, string $columns ): array {
+		$table_name = esc_sql( $table_name );
+		$columns    = esc_sql( $columns );
+
+		// Add index only if it does not exist.
+		if ( $this->has_index( $index_name ) ) {
+			return $results;
+		}
+
+		$sql = "ALTER TABLE `{$table_name}` ADD INDEX `{$index_name}` ( {$columns} )";
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery,WordPress.DB.PreparedSQL.NotPrepared
+		$updated = $wpdb->query( $sql );
+
+		$message = $updated ?
+			sprintf( 'Added index to the %s table on %s.', $table_name, $columns ) :
+			sprintf( 'Failed to add an index on the %s table for %s.', $table_name, $columns );
+
+		$results[ "{$table_name}.{$columns}" ] = $message;
+		return $results;
+	}
+
+	/**
+	 * Adds foreign key constraints to a table if they don't already exist.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $table_name        The name of the table to add the foreign key to.
+	 * @param string $foreign_key_name  The name of the foreign key constraint.
+	 * @param string $column_name       The column that references the foreign key.
+	 * @param string $referenced_table  The referenced table name.
+	 * @param string $referenced_column The referenced column in the foreign table.
+	 * @param string $on_delete_action  The action on delete (e.g., CASCADE).
+	 *
+	 * @return void
+	 */
+	protected function add_foreign_key(
```

Comment:
I wouldn't use forgeign keys at all. Its not our DBs so its hard to support customers already. WP doesn't have a system on its own using foreign keys so i would strongly advice against using them as well.